Add de.peterge.combined

peterge1998 opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

hfiguiere commented:
you should make sure it's only built for x86_64. See https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

hfiguiere reviewed (changes requested):
you are missing an appdata file.

hfiguiere commented on de.peterge.combined.yml:
```diff
@@ -0,0 +1,32 @@
+command: "Combined!.x86_64"
+
+finish-args:
+  - --socket=x11
```
you need `- --share=ipc` for X11. And I suspect you also want `- --device=dri` for openGL.

peterge1998 commented on de.peterge.combined.yml:
```diff
@@ -0,0 +1,32 @@
+      - install -D "Combined!.pck" "/app/bin/Combined!.pck"
+      - install -Dm644 icon.png -t /app/bin/icon.png
+      - install -Dm644 icon.png -t /app/share/icons/hicolor/256x256/apps/
+      - install -Dm644 de.peterge.combined.desktop /app/share/applications/de.peterge.comdined.desktop
```
I don't get this. What is correct here?
`      - install -Dm644 icon.png -t /app/bin/`

hfiguiere replied:
definitely not.

the `.desktop` file has `Icon=icon.png` and that's wrong.  It should be `Icon=icon` (and `rename-icon` take care of the rest), to match with what you installed.

peterge1998 commented:
> you should make sure it's only built for x86_64. See https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64
Added a basic flathub.json in 54d9ade2b29eaa5bc99529400aa3f0f2e5f646a9.

>You need a rename-icon: icon
Added rename-icon: icon in e8682113ad979f496cc6f8825ea0836b0b0eb5a7.

And the OpenGL/x11 lines in e9c6287e053c894e32ce6525715bb07e3b7e6348.

Sorry, but I am new to publishing on Flathub ^^

hfiguiere commented:
still missing the appdata file.

nedrichards commented:
bot, build de.peterge.combined

(this will fail, as @hfiguiere says, but it'll help you see the process and you can call the command when you push a new commit to check it)

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
Queued test build for de.peterge.combined.

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
What is going wrong here? [The `.desktop` file](https://gitlab.com/peterge/combined-assets/-/blob/main/de.peterge.combined.desktop)

peterge1998 commented:
bot, build de.peterge.combined

hfiguiere commented on de.peterge.combined.metainfo.xml:
```diff
@@ -0,0 +1,28 @@
+
+<?xml version="1.0" encoding="UTF-8"?>
```
this file isn't installed. hence the build failure.

hfiguiere commented on de.peterge.combined.metainfo.xml:
```diff
@@ -0,0 +1,28 @@
+<component type="game">
+  <id>com.valvesoftware.SteamLink</id>
+  <launchable type="desktop-id">de.peterge.combined</launchable>
+  <metadata_license>MIT</metadata_license>
```
we prefer CC0-1.0 for the metadata license.

hfiguiere commented on de.peterge.combined.metainfo.xml:
```diff
@@ -0,0 +1,28 @@
+
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="game">
+  <id>com.valvesoftware.SteamLink</id>
```
the ID doesn't match.

peterge1998 commented:
bot, build de.peterge.combined

hfiguiere commented on de.peterge.combined.yml:
```diff
@@ -0,0 +1,35 @@
+      - install -Dm644 icon.png -t /app/bin/icon.png
+      - install -Dm644 icon.png -t /app/share/icons/hicolor/256x256/apps/
+      - install -Dm644 de.peterge.combined.desktop /app/share/applications/de.peterge.combined.desktop
+      - install -Dm644 de.peterge.combined.metainfo.xml /app/share/de.peterge.combined.metainfo.xml
```
```suggestion
      - install -Dm644 de.peterge.combined.metainfo.xml /app/share/metainfo/de.peterge.combined.metainfo.xml
```

hfiguiere replied:
but the file isn't listed below in the `sources` so it won't work until it is.

peterge1998 commented:
bot, build de.peterge.combined

hfiguiere commented:
you can't install a file that's not listed in the sources.

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
![image](https://user-images.githubusercontent.com/47355238/144508092-0300a684-12f5-4f0a-ad0a-3ddf2978f950.png)
Okay, finally the page in gnome's software looks good. One last thing: it does not display my screenshot, but maybe this does not work for the beta build yet. Finally ready to become offical available on flathub :rocket:

peterge1998 commented:
bot, build de.peterge.combined

peterge1998 commented:
I did not notice this error, which prevents the app from running, on a wayland session yet: `initialize: X11 Display is not available`
Is there a way how i can run my app on wayland? If i run the .x86_64 file on a wayland session directly, it starts fine. What is the best way do define it with `--socket=fallback-x11`?

peterge1998 commented:
bot, build de.peterge.combined

barthalion commented:
/merge
